AlfrescoApp/Model/Managers/RealmSyncManager+Refresh.m

```diff
@@ -0,0 +1,217 @@
+/*******************************************************************************
+ * Copyright (C) 2005-2016 Alfresco Software Limited.
+ *
+ * This file is part of the Alfresco Mobile iOS App.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+
+#import "RealmSyncManager+Refresh.h"
+#import "RealmSyncManager+Internal.h"
+#import "AlfrescoNode+Sync.h"
+
+@implementation RealmSyncManager (Refresh)
+
+#pragma mark - Public Methods
+
+- (void)refreshWithCompletionBlock:(void (^)(BOOL completed))completionBlock
+{
+    // STEP 1 - Mark all top level nodes as pending sync status
+    [self markTopLevelNodesAsPending];
+    
+    // STEP 2 - Invalidate the parent/child hierarchy in the database.
+    [self invalidateParentChildHierarchy];
+    
+    // STEP 3 - Recursively request child folders from the server and update the database structure to match
+    [self rebuildDataBaseWithCompletionBlock:^(BOOL completed)
+    {
+        // STEP 4 - Remove any synced content that is no longer within a sync set
+        [self cleanDataBaseOfUnwantedNodes];
+        
+        // STEP 5 - Start downloading any new content that appears inside the sync set
+        [self startDownloadingNewContent];
+        
+        if (completionBlock)
+        {
+            completionBlock(YES);
+        }
+    }];
+}
+
+#pragma mark - Private Methods
+
+- (void)markTopLevelNodesAsPending
+{
+    RLMRealm *realm = self.mainThreadRealm;
+    // Get all top level nodes.
+    RLMResults *allTopLevelNodes = [[RealmManager sharedManager] topLevelNodesInRealm:realm];
+    
+    // Mark all top level nodes as "pending sync" status.
+    for (RealmSyncNodeInfo *node in allTopLevelNodes)
+    {
+        SyncNodeStatus *nodeStatus = [[RealmSyncManager sharedManager] syncStatusForNodeWithId:node.syncNodeInfoId];
+        nodeStatus.status = SyncStatusWaiting;
+    }
+}
+
+- (void)invalidateParentChildHierarchy
+{
+    RLMRealm *realm = self.mainThreadRealm;
+    RLMResults *allNodes = [[RealmManager sharedManager] allNodesInRealm:realm];
+    
+    for (RealmSyncNodeInfo *node in allNodes)
+    {
+        if (node.parentNode)
+        {
+            [realm beginWriteTransaction];
+            node.parentNode = nil;
+            [realm commitWriteTransaction];
```

Comment:
Since beginWriteTransaction and commitWriteTransaction involve writing changes to realm files (non-trivial work) please move the calls to enclose the for loop (such that there would be only one write transaction for the entire for loop)